Update cli.md

sleepyhollo opened the pull request:
closes #844
Add a JSON config file example for the CLI

Outstanding questions
- do I need to write and/or run tests for the contribution to documentation?
- how would I add non-json type values in a json config file? For example, [PyMdown Extensions][pymdownx] [emoji][pymdownx-emoji] extension has an option `emoji_index` which seems to take a reference to a function or a module as its input. This can be seen in the [Github-ish Configurations][pymdownx-faq] FAQ in PyMdown's docs. I tried passing `"emoji_index": "pymdownx.emoji.gemoji"` through a json config file, but that failed. So perhaps thats a limitation of the json configs file...

[pymdownx]: https://facelessuser.github.io/pymdown-extensions/
[pymdownx-emoji]: https://facelessuser.github.io/pymdown-extensions/extensions/emoji/
[pymdownx-faq]: https://facelessuser.github.io/pymdown-extensions/faq/

sleepyhollo committed:
Update cli.md

closes #844
Add a JSON config file example for the CLI

waylan commented on docs/cli.md:
````diff
@@ -160,6 +160,18 @@ myext:
     option2: True
 ```
 
+Similarly, a [JSON] config file might look like this:
````
`config` is [not a properly spelled word][1]. Please use the full word: `configuration`.

[1]: https://travis-ci.org/Python-Markdown/markdown/jobs/558492124#L310

waylan commented:
> do I need to write and/or run tests for the contribution to documentation?

No, that's not necessary. Or at least we never have.

> how would I add non-json type values in a json config file?

You don't. Which is why we also support YAML. YAML has specific support for specifying Python types. In fact, JSON is a limited subset of YAML.

sleepyhollo committed:
Update cli.md

changed config to configuration

sleepyhollo committed:
Update cli.md

Added a line mentioning that YML file is required for non-json type option values

sleepyhollo committed:
Update cli.md

fix typo. remove json link.
